File: react-type-vite/src/services/api/response/studentResponse.ts
import type { DepartmentResponse } from "./departmentResponse";
import type { EducationalUnitResponse } from "./educationalUnitResponse";

export interface StudentResponse {
  id: string;
  username: string;
  email: string;
  firstName: string;
  lastName: string;
  dob?: string;
  studentId: string;
  departmentId?: string;
  educationalUnitId?: string;
  description?: string;
  socialUrl?: string;
  className?: string;
  department?: DepartmentResponse;
  educationalUnit?: EducationalUnitResponse;
  submittedAssignments: number
  totalAssignments: number
  completedQuizzes: number
  totalQuizzes: number
  averageScore: number
  lastAccessTime?: Date
  totalLearningHours: number
  enrollmentDate?: Date
  accountStatus: string
}

File: spring-microservices/course-service/src/main/java/com/hoangphihiep/dto/response/StudentResponse.java
package com.hoangphihiep.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class StudentResponse {

    private String id;

    private String username;

    private String email;

    private String firstName;

    private String lastName;

    private String dob;

    private String studentId;

    private String departmentId;

    private String educationalUnitId;

    private String description;

    private String socialUrl;

    private String className;

    private DepartmentResponse department;

    private EducationalUnitResponse educationalUnit;

    private Integer submittedAssignments;

    private Integer totalAssignments;

    private Integer completedQuizzes;

    private Integer totalQuizzes;

    private Integer averageScore;

    private String accountStatus;

    private String avatarUrl;
}

